### ISSUE TITLE
[Repo Rearchitecture] Define how to determine a quickstart/install plan has been updated

### ISSUE THREAD

**Issue by jpvajda** [newrelic/newrelic-quickstarts#631]
In our current approach to the repo re-archiecture work we wil be making  400+ calls to the catalog API for every quickstart that exists in the repo when a update occurs and this seems inefficient.  Should we be able to determine which quickstart/install plan has actually changed and only make a call for that particular component to reduce load and requests on the catalog service?

### Acceptance Criteria

- [x] Discuss with the Virtuoso + DevEn teams this concern
- [x] Determine if we should consider a more elegant approach 
- [ ] OR if they aren't concerned, then go with the simple approach

**Comment by jpvajda** [comment]
@rudouglas this came up today in our backlog review, so I was going to follow up on it.

**Comment by jpvajda** [comment]
I've asked @jerelmiller and @jbeveland27 their opinions on this in our epic planning doc.

**Comment by jpvajda** [comment]
from @jerelmiller  

_Wouldn't this be a result of a PR being opened against the `main` branch? My thinking was that once a PR is merged, the GitHub action would send the mutation for each quickstart that was changed in the PR. As long as the file for that quickstart includes an ID, this would allow you to determine whether to create or update it.

Was this how you were thinking of it? This question makes me think that you're considering sending every quickstart through the mutations when merging a PR.

I'd love to get a better idea of what you're thinking here as well._

**Comment by rudouglas** [comment]
Ye I think the best approach here would be to read the PR to determine what quickstarts have changes and as Jerel said, determine whether it's new/updated by checking for the existence of the `id` in the config file.

This is semi related, but we currently see a lot of contributors copy/pasting `id` from another quickstart when creating a new one, and we check whether they've done this manually, so i'll create a ticket to check for this on a PR

**Comment by rudouglas** [comment]
@jpvajda for adding labels `new-quickstart` and `improved-quickstarts` we use an action that checks for `files_added` and `files_modified` on a PR so we could use this or code our own solution if we don't want to rely on a 3rd party

**Comment by rudouglas** [comment]
This should also include Install Plans

**Comment by jpvajda** [comment]
@rudouglas Elementary my dear Watson. 🕵️  We can close this out as I see you've updated other tickets this relates to.

**Comment by jerelmiller** [comment]
@jpvajda @rudouglas Just an FYI we had a meeting last week to chat about some complexities with syncing between environments. tl;dr; we landed on a couple decisions:

* We are going to "switch back" to let the client generate the UUID
* We are going to consolidate from 2 mutations (`createQuickstart`, `updateQuickstart`) to a single mutation (`submitQuickstart`)

Being that we need the quickstart in all 3 environments (staging, prod, eu), we determined it would be pretty difficult to try and sync that ID across all 3 envs when submitting a quickstart since the initial proposal was to have the db generate the ID.

Not sure what ticket you opened from here, but I'd recommend getting in touch with @aswanson-nr to chat about this.
